### PR TITLE
create vpc

### PR DESCRIPTION
**Pull request by marcmael1**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+resource "aws_route_table_association" "private0" {
+  subnet_id      = aws_subnet.private0.id
+  route_table_id = aws_route_table.private0.id
+}
+
+resource "aws_route_table_association" "private1" {
+  subnet_id      = aws_subnet.private1.id
+  route_table_id = aws_route_table.private1.id
+}
```
Comment (vladimirmukhin) [inline main.tf]: Each file should end with an empty line aka trailing newline

File: provider.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
```
Comment (vladimirmukhin) [inline provider.tf]: Please remove this comment. It's too obvious and don't provide any value

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,138 @@
+# Create a VPC
```
Comment (vladimirmukhin) [inline main.tf]: Please remove this comment. It's too obvious and don't provide any value

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+resource "aws_vpc" "demo_vpc" {
+  cidr_block = "10.0.0.0/16"
+  tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  cidr_block = "10.0.0.0/16"
  
  tags = {
```

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  vpc_id = aws_vpc.demo_vpc.id
+
+  tags = {
+    Name = "demo-gw"
+  }
+}
+
+resource "aws_eip" "nat0" {
+  vpc = true
+  tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  vpc = true
  
  tags = {
```

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    "Name" = "demo-nat0"
+  }
+}
+
+resource "aws_eip" "nat1" {
+  vpc = true
+  tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  vpc = true
  
  tags = {
```

File: main.tf [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    "Name" = "demo-nat1"
+  }
+}
+
+resource "aws_nat_gateway" "nat-gw0" {
+  allocation_id = aws_eip.nat0.id
+  subnet_id     = aws_subnet.public0.id
+  tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  subnet_id     = aws_subnet.public0.id
  
  tags = {
```

File: main.tf [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    "Name" = "demo-nat-gw0"
+  }
+}
+
+resource "aws_nat_gateway" "nat-gw1" {
+  allocation_id = aws_eip.nat1.id
+  subnet_id     = aws_subnet.public1.id
+   tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  subnet_id     = aws_subnet.public1.id
  
   tags = {
```

File: main.tf [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+resource "aws_route_table" "public" {
+  vpc_id = aws_vpc.demo_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.demo-gw.id
+  }
+   tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  }
  
   tags = {
```

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+resource "aws_route_table" "private0" {
+  vpc_id = aws_vpc.demo_vpc.id
+
+  route {
+    cidr_block     = "0.0.0.0/0"
+    nat_gateway_id = aws_nat_gateway.nat-gw0.id
+  }
+   tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  }
  
   tags = {
```

File: main.tf [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+resource "aws_route_table" "private1" {
+  vpc_id = aws_vpc.demo_vpc.id
+
+  route {
+    cidr_block     = "0.0.0.0/0"
+    nat_gateway_id = aws_nat_gateway.nat-gw1.id
+  }
+   tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  }
  
   tags = {
```

File: main.tf [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+  route_table_id = aws_route_table.public.id
+}
+
+resource "aws_route_table_association" "public1" {
+  subnet_id      = aws_subnet.public1.id
+  route_table_id = aws_route_table.public.id
+}
+
+
+resource "aws_route_table_association" "private0" {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
}

resource "aws_route_table_association" "private0" {
```

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]
Please run "terraform fmt" so terraform will automatically adjust some style fixes into your code. After that, commit the result.
More info about terraform fmt here: https://www.terraform.io/cli/commands/fmt

File: main.tf [hunk 13]
```diff
@@ -0,0 +1,117 @@
+resource "aws_vpc" "demo_vpc" {
+  cidr_block = "10.0.0.0/16"
+  tags = {
+    "Name" = "demo-vpc"
+  }
+}
+resource "aws_subnet" "public0" {
```
Comment (vladimirmukhin) [inline main.tf]: Top-level blocks should always be separated from one another by one blank line. 

```suggestion
}

resource "aws_subnet" "public0" {
```

Comment (vladimirmukhin) [reply main.tf]: Please adjust it here and for all other top-level blocks

File: main.tf [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+  cidr_block = "10.0.1.0/24"
+
+  tags = {
+    Name = "public-subnet1"
+  }
+}
+resource "aws_subnet" "private0" {
+  vpc_id     = aws_vpc.demo_vpc.id
+  cidr_block = "10.0.2.0/24"
+  tags = {
```
Comment (vladimirmukhin) [inline main.tf]: Nested blocks should also be separated by blank lines
```suggestion
  cidr_block = "10.0.2.0/24"
  
  tags = {
```

Comment (vladimirmukhin) [reply main.tf]: Please adjust it here and for all other nested blocks

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: main.tf (hunk 3 above)
Comment (marcmael1) [reply main.tf]: did removed it

File: main.tf (hunk 5 above)
Comment (marcmael1) [reply main.tf]: Added space

File: main.tf (hunk 8 above)
Comment (marcmael1) [reply main.tf]: Fixed it

File: main.tf (hunk 11 above)
Comment (marcmael1) [reply main.tf]: Fixed it

File: main.tf (hunk 14 above)
Comment (marcmael1) [reply main.tf]: Fixed it

File: main.tf (hunk 13 above)
Comment (marcmael1) [reply main.tf]: ok

File: main.tf (hunk 12 above)
Comment (marcmael1) [reply main.tf]: Done

**Review by vladimirmukhin — APPROVED** [review]

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**